py/server/deephaven_internal/auto_completer/_completer.py

```diff
@@ -253,3 +258,36 @@ def do_hover(
             hoverstring += '\n---\n' + wrap_plaintext(raw_docstring)
 
         return hoverstring.strip()
+
+    def check_recursion_limit(self):
+        """
+        Tests for python+jedi+numpy versions that are susceptible to a RecursionError/segfault issue, and lowers
+        the recursion limit, warning if the limit is raised externally
+        """
+        if not sys.version.startswith('3.9.') and not sys.version.startswith('3.10.'):
```

Comment:
I think the more canonical way to do this would be:

```python
sys.version_info < (3, 9) || sys.version_info >= (3, 11)
```

Reply:
Ah that's handy, I'll do that (though with `or`, since `||` is much harder to read than `>=` or other operators...)